src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,148 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.prev = prev;
+            this.item = element;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<T>(last, value, null);
+        if (last != null) {
+            last.next = node;
+        }
+        last = node;
+        if (first == null) {
+            first = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index does not exist");
+        }
+        Node<T> newNode = new Node<>(null, value, null);
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
```

Comment:
let's move this first `if` to the start of this method and write `return;` in it